speed.py

```diff
@@ -0,0 +1,111 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+# from datasets import load_dataset
+
+# dataset = load_dataset("roszcz/internship-midi-data-science", split="train")
+
+
+"""
+SPEED
+"""
+
+
+def get_time_vs_speed(record):
+    df = pd.DataFrame(record["notes"])
+
+    start = df.start.values
+    end = df.end.values
+
+    if (max(end) - min(start)) > 120:
+        step = 60
+        unit = "min"
+    else:
+        step = 1
+        unit = "sec"
+
+    notes_in_time = pd.DataFrame(
+        df.groupby(
+            pd.cut(
+                df["start"], range(int(min(df["start"])), int(max(df["start"])), step)
+            )
+        ).pitch.count()
+    )
+    notes_in_time.index = np.arange(
+        int(min(start)) + 1, len(notes_in_time) + int(min(start)) + 1
+    )
+    notes_in_time.index.name = f"{unit}_number"
+    notes_in_time.rename(columns={"pitch": "number_of_notes_played"}, inplace=True)
+
+    return notes_in_time
+
+
+def plot_chart_time_vs_speed(notes_in_time):
+    unit = notes_in_time.index.name.split("_")[0]
+    step = 1 if unit == "sec" else 60
+
+    f, ax = plt.subplots(1, 1, figsize=[14, 6])
+    ax.scatter(
+        x=notes_in_time.index,
+        y=notes_in_time["number_of_notes_played"].values / step,
+    )
```

Comment:
Scatter plot works, but I'd suggest a regular plot:

```python
ax.plot(notes_in_time.time, notes_in_time.notes_per_second)
```

I think this should give a better idea about continuity of the measured process

Reply:
![image](https://github.com/Nospoko/midi-internship/assets/117731792/3b5dd9f8-c859-4d00-a6ce-b0f962721c03)

I fixed the plot, hope it is better now :)